core/src/impl/Kokkos_FunctorAdapter.hpp

```diff
@@ -2091,13 +2091,33 @@ struct FunctorFinal<FunctorType, ArgTag,
                     decltype(
                         FunctorFinalFunction<FunctorType, ArgTag>::enable_if(
                             &FunctorType::final))> {
-  KOKKOS_FORCEINLINE_FUNCTION static void final(const FunctorType& f, void* p) {
+  template <typename Dummy = ArgTag>
+  KOKKOS_FORCEINLINE_FUNCTION static std::enable_if_t<
+      std::is_same<Dummy, void>::value>
```

Comment:
This works but departs from what is done above for `FunctorValueInit` and `FunctorValueJoin` where two separate specializations are provided.

Reply:
Yes, this seems to be less code so I went with this approach.

Reply:
If you insist, I can change, of course.

Reply:
I would prefer consistency within that file but I will not block if I am the only one feeling that way.